components/fxa-client/src/telemetry.rs

```diff
@@ -2,12 +2,268 @@
+            .returns_once(Ok(Some(ResponseAndETag {
+                response: ProfileResponse {
+                    uid: "12345ab".to_string(),
+                    email: "foo@bar.com".to_string(),
+                    display_name: None,
+                    avatar: "https://foo.avatar".to_string(),
+                    avatar_default: true,
+                    ecosystem_anon_id: Some(ecosystem_anon_id.to_owned()),
+                },
+                etag: None,
+            })));
+        fxa.set_client(Arc::new(client));
+
+        assert_eq!(fxa.get_ecosystem_anon_id().unwrap(), ecosystem_anon_id);
+    }
+
+    #[test]
+    fn get_ecosystem_anon_id_generate_anon_id() {
+        let mut fxa = fxa_setup();
+
+        let mut client = FxAClientMock::new();
+        client
+            .expect_profile(
+                mockiato::Argument::any,
+                |token| token.partial_eq("profiletok"),
+                mockiato::Argument::any,
+            )
+            .times(1)
+            .returns_once(Ok(Some(ResponseAndETag {
+                response: ProfileResponse {
+                    uid: "12345ab".to_string(),
+                    email: "foo@bar.com".to_string(),
+                    display_name: None,
+                    avatar: "https://foo.avatar".to_string(),
+                    avatar_default: true,
+                    ecosystem_anon_id: None,
+                },
+                etag: None,
+            })));
+        client
+            .expect_fxa_client_configuration(mockiato::Argument::any)
+            .times(1)
+            .returns_once(Ok(ClientConfigurationResponse {
+                auth_server_base_url: "https://foo.bar".to_owned(),
+                oauth_server_base_url: "https://foo.bar".to_owned(),
+                profile_server_base_url: "https://foo.bar".to_owned(),
+                sync_tokenserver_base_url: "https://foo.bar".to_owned(),
+                ecosystem_anon_id_keys: Some(vec![Jwk {
+                    kid: Some("LlU4keOmhTuq9fCNnpIldYGT9vT9dIDwnu_SBtTgeEQ".to_owned()),
+                    key_parameters: JwkKeyParameters::EC(ECKeysParameters {
+                        crv: "P-256".to_owned(),
+                        x: "i3FM3OFSCZEoqu-jtelXwKt6AL4ODQ75NUdHbcLWQSo".to_owned(),
+                        y: "nW-S3QiHDo-9hwfBhKnGKarkt_PVqVyIPUytjutTunY".to_owned(),
+                    }),
+                }]),
+            }));
+        client
+            .expect_ecosystem_anon_id(
+                mockiato::Argument::any,
+                mockiato::Argument::any,
+                mockiato::Argument::any,
+            )
+            .times(1)
+            .returns_once(Ok(()));
+        fxa.set_client(Arc::new(client));
+
+        let ecosystem_anon_id = fxa.get_ecosystem_anon_id().unwrap();
+        // Well, it looks like a jwe folks.
+        assert!(ecosystem_anon_id.chars().filter(|c| c == &'.').count() == 4);
+        assert!(fxa.state.ecosystem_user_id.unwrap().len() == 32);
```

Comment:
For bonus points, you could use a known JWK public key in the mock config above, and use the corresponding known private key to decrypt it here. Not sure how much value that would provide in practice though.

Reply:
Yeah I was going to do that then I felt like it's just re-testing the jw stack.